modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/ApplicationPolicyRedeploymentTestCase.java

```diff
@@ -66,37 +71,68 @@ public void objectStoreSamePartitionWorksAfterRedeployingPolicy() throws Excepti
 
     executeApplicationFlow(FLOW_NAME);
   }
 
-  private PolicyFileBuilder policyWithPlugin() {
-    Map<String, Object> map = new HashMap<>();
-    map.put("store", "");
-    MulePolicyModelBuilder mulePolicyModelBuilder = new MulePolicyModelBuilder()
-        .setMinMuleVersion(MIN_MULE_VERSION).setName(OS_POLICY_NAME)
-        .setRequiredProduct(Product.MULE)
-        .withBundleDescriptorLoader(createBundleDescriptorLoader(OS_POLICY_NAME, MULE_POLICY_CLASSIFIER,
-                                                                 PROPERTIES_BUNDLE_DESCRIPTOR_LOADER_ID))
-        .withClassLoaderModelDescriptorLoader(new MuleArtifactLoaderDescriptor(MULE_LOADER_ID, map));
-
-    return new PolicyFileBuilder(OS_POLICY_NAME)
-        .describedBy(mulePolicyModelBuilder.build());
+  @Test
+  @Issue("W-13563214")
+  @Description("There was an issue with the inner fluxes of an async operation in a policy being disposed (and never rebuilt) after another operation policy was undeployed/redeployed and the CompositeOperationPolicy sinks were completed")
+  public void whenRedeployingOperationPolicyThenOtherOperationPoliciesWithAsyncOperationsStillWork() throws Exception {
+    // We will be deploying two policies and redeploying one of them
+    // It is important that the policy that remains deployed has an asynchronous operation in order to reproduce the issue
+    PolicyFileBuilder policyToRedeployFileBuilder = createPolicyFileBuilder(POLICY_WITH_OPERATION_NAME);
+    PolicyFileBuilder policyToKeepFileBuilder = createPolicyFileBuilder(POLICY_WITH_ASYNC_OPERATION_NAME);
+
+    PolicyParametrization policyToRedeployParametrization = getParametrization(policyToRedeployFileBuilder, 1);
+    PolicyParametrization policyToKeepParametrization = getParametrization(policyToKeepFileBuilder, 2);
+
+    policyManager.registerPolicyTemplate(policyToRedeployFileBuilder.getArtifactFile());
+    policyManager.registerPolicyTemplate(policyToKeepFileBuilder.getArtifactFile());
+
+    ApplicationFileBuilder applicationFileBuilder = createExtensionApplicationWithServices(APP_WITH_EXTENSION_PLUGIN_CONFIG,
+                                                                                           helloExtensionV1Plugin);
+    addPackedAppFromBuilder(applicationFileBuilder);
+
+    startDeployment();
+    assertApplicationDeploymentSuccess(applicationDeploymentListener, applicationFileBuilder.getId());
+
+    policyManager.addPolicy(applicationFileBuilder.getId(), policyToRedeployFileBuilder.getArtifactId(),
+                            policyToRedeployParametrization);
+    policyManager.addPolicy(applicationFileBuilder.getId(), policyToKeepFileBuilder.getArtifactId(), policyToKeepParametrization);
+
+    assertManualExecutionsCount(2);
+
+    policyManager.removePolicy(applicationFileBuilder.getId(), policyToRedeployFileBuilder.getArtifactId());
+
+    // TODO: actually wait until the stale policy is disposed to avoid a source of test flakiness
+    sleep(2000);
```

Comment:
Will try to spend a little bit more time trying to solve this before merging the PR.